packages/api/tests/models/test-executions-model.js

```diff
@@ -1,148 +1,344 @@
-  executionModel.generateDocFromPayload = (_payload) => returnDoc(arn, executionStatus);
-  await executionModel.createExecutionFromSns({ payload: originalPayload });
-}
+const createExecutionMessage = ({
+  executionName,
+  status = 'running'
+}) => ({
+  cumulus_meta: {
+    execution_name: executionName,
+    state_machine: stateMachineArn,
+    workflow_start_time: Date.now()
+  },
+  meta: {
+    collection: {
+      name: randomString(),
+      version: randomString()
+    },
+    provider: {
+      host: randomString(),
+      protocol: 's3'
+    },
+    workflow_name: 'test',
+    status,
+    workflow_tasks: {
+      task1: {
+        arn: randomString(),
+        name: randomString(),
+        version: 1
+      }
+    }
+  }
+});
 
 test.before(async () => {
   process.env.ExecutionsTable = randomString();
-  manager = new Manager({
-    tableName: process.env.ExecutionsTable,
-    tableHash: { name: 'arn', type: 'S' },
-    schema: schemas.execution
-  });
-  await manager.createTable();
+  executionModel = new Execution();
+  await executionModel.createTable();
 });
 
 test.after.always(async () => {
-  await manager.deleteTable();
+  await executionModel.deleteTable();
+});
+
+test.beforeEach(async (t) => {
+  t.context.executionName = randomString();
+  t.context.arn = getExecutionArn(t.context.executionName);
+  t.context.executionUrl = getExecutionUrl(t.context.arn);
+});
+
+test.afterEach.always(async (t) => {
+  await executionModel.delete({ arn: t.context.arn });
+});
+
+test.serial('generateRecord() using payload without cumulus_meta.state_machine throws error', async (t) => {
+  await t.throwsAsync(
+    () => Execution.generateRecord({
+      cumulus_meta: {
+        execution_name: randomString()
+      }
+    })
+  );
+});
+
+test.serial('generateRecord() using payload without cumulus_meta.execution_name throws error', async (t) => {
+  await t.throwsAsync(
+    () => Execution.generateRecord({
+      cumulus_meta: {
+        state_machine: randomString()
+      }
+    })
+  );
+});
+
+test('generateRecord() generates the correct record for a running execution', async (t) => {
+  const { arn, executionName, executionUrl } = t.context;
+  const message = createExecutionMessage({
+    executionName
+  });
+  message.payload = { foo: 'bar' };
+
+  const record = await Execution.generateRecord(message);
+
+  const { collection } = message.meta;
+
+  t.is(record.name, executionName);
+  t.is(record.arn, arn);
+  t.is(record.status, 'running');
+  t.is(record.execution, executionUrl);
+  t.is(record.type, message.meta.workflow_name);
+  t.is(record.createdAt, message.cumulus_meta.workflow_start_time);
+  t.is(record.collectionId, constructCollectionId(collection.name, collection.version));
+  // Seems like this is wrong. If message.exception is undefined, then record.error
+  // should be undefined or an empty object?
+  t.deepEqual(record.error, { Error: 'Unknown Error', Cause: undefined });
```

Comment:
I think our handling of Cumulus message exceptions is busted, but didn't want to add that to this PR